File: frontend/src/services/AppointmentService.ts
import axiosClient from "../api/axiosClient";

const AppointmentService = {
    createAppointment: (data) => axiosClient.post("/appointments",data),
    getTodayAppointmentsByDoctor: () => axiosClient.get("/doctors/me/appointments/today"),
    getPatientAppointments: (pageNumber,pageSize,sort="appointmentDate,desc") => axiosClient.get("/patients/me/appointments",{
        params:{
            pageNumber,
            pageSize,
            sort
        }
    }),
    cancelAppointment: (id) => axiosClient.put(`/patients/me/appointments/${id}`),
    getPatientAppointmentById: (id) => axiosClient.get("/patient/me/appointments/"+id),
    getAppointmentsByDoctor: (pageNumber,pageSize,sort ="appointmentDate,desc") => axiosClient.get(`/doctors/me/appointments?pageNumber=${pageNumber}&pageSize=${pageSize}&sort=${sort}`),
    getAppointmentDetailByDoctor: (id) => axiosClient.get("/doctors/me/appointments/"+id)
}

export default AppointmentService;

File: backend/src/main/java/com/example/doctor/appointment/service/AppointmentService.java
package com.example.doctor.appointment.service;

import com.example.doctor.appointment.dto.AppointmentDTO;
import com.example.doctor.appointment.dto.DoctorAppointmentDTO;
import com.example.doctor.appointment.dto.ResponseDTO;
import com.example.doctor.appointment.entity.Appointment;
import org.springframework.data.domain.Page;
import org.springframework.security.core.Authentication;

import java.time.LocalDate;
import java.util.List;

public interface AppointmentService {
    ResponseDTO<AppointmentDTO> createAppointment(AppointmentDTO request);
    ResponseDTO<Page<AppointmentDTO>> findAppointmentsByPatient(int pageNumber,int pageSize,String sort);
    ResponseDTO<AppointmentDTO> getAppointment(Long appointmentId);
    ResponseDTO<String> cancelAppointment(Long appointmentId);
    ResponseDTO<Page<DoctorAppointmentDTO>> getAppointmentsByDoctor(int page,int pageSize,String sort);
    ResponseDTO<DoctorAppointmentDTO> getAppointmentDetailByDoctor(Long appointmentId);
    ResponseDTO<List<DoctorAppointmentDTO>> getTodayAppointmentsByDoctor();
}
